code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -110,12 +110,23 @@
 		if(!isliving(M))
 			return
 		var/impact_message = ""
+		if(isXeno(M))
```

Comment:
This would be ideally handled by procs specific to each mob, something like `on_hit`, instead of a long list of checks and exceptions.